docker/docker-entrypoint.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+set -Eeo pipefail
+
+source /usr/local/bin/upstream-docker-entrypoint.sh
+
+# sync $POSTGRES_PASSWORD to supabase-specific roles
+pg_sync_password() {
+  # PGPASSWORD is required for psql when authentication is required for 'local' connections via pg_hba.conf and is otherwise harmless
+  # e.g. when '--auth=md5' or '--auth-local=md5' is used in POSTGRES_INITDB_ARGS
+	export PGPASSWORD="${PGPASSWORD:-$POSTGRES_PASSWORD}"
+	docker_temp_server_start "$@"
+
+	# alter the supabase_admin password
+	docker_process_sql <<-'EOSQL'
+		\set pgpass `echo "$POSTGRES_PASSWORD"`
+		ALTER USER supabase_admin WITH PASSWORD :'pgpass';
+	EOSQL
+
+	# execute the roles SQL file using docker_process_sql
+	docker_process_sql -f /docker-entrypoint-initdb.d/init-scripts/99-roles.sql
```

Comment:
@jgoux let's make this path an env var and only run pg_sync_password if this path is defined.